targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/gpio_irq_api.c

```diff
@@ -0,0 +1,241 @@
+                case 7: channels_irq = &gpio0_irq7; break;
+                case 8: channels_irq = &gpio0_irq8; break;
+                case 9: channels_irq = &gpio0_irq9; break;
+                case 10: channels_irq = &gpio0_irq10; break;
+                case 11: channels_irq = &gpio0_irq11; break;
+                case 12: channels_irq = &gpio0_irq12; break;
+                case 13: channels_irq = &gpio0_irq13; break;
+                case 14: channels_irq = &gpio0_irq14; break;
+                case 15: channels_irq = &gpio0_irq15; break;
+                case 16: channels_irq = &gpio1_irq0; break;
+                case 17: channels_irq = &gpio1_irq1; break;
+                case 18: channels_irq = &gpio1_irq2; break;
+                case 19: channels_irq = &gpio1_irq3; break;
+                case 20: channels_irq = &gpio1_irq4; break;
+                case 21: channels_irq = &gpio1_irq5; break;
+                case 22: channels_irq = &gpio1_irq6; break;
+                case 23: channels_irq = &gpio1_irq7; break;
+                case 24: channels_irq = &gpio1_irq8; break;
+                case 25: channels_irq = &gpio1_irq9; break;
+                case 26: channels_irq = &gpio1_irq10; break;
+                case 27: channels_irq = &gpio1_irq11; break;
+                case 28: channels_irq = &gpio1_irq12; break;
+                case 29: channels_irq = &gpio1_irq13; break;
+                case 30: channels_irq = &gpio1_irq14; break;
+                case 31: channels_irq = &gpio1_irq15; break;
+
+        }
+        NVIC_SetVector((IRQn_Type)(PININT_IRQ + obj->ch), (uint32_t)channels_irq);
+        NVIC_EnableIRQ((IRQn_Type)(PININT_IRQ + obj->ch));
+
+        return 0;
+    }
+}
+
+void gpio_irq_free(gpio_irq_t *obj) {
+}
+
+void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable) {
+        unsigned int ch_bit = (1 << obj->ch);
+
+        // Clear interrupt
+        if (obj->ch <16)
+        {
+                if (!(CMSDK_GPIO_0->INTTYPESET & ch_bit))
+                {
+                    CMSDK_GPIO_0->INTCLEAR = ch_bit;
+                }
+        }
+        if (obj->ch >= 16)
+        {
+                if (!(CMSDK_GPIO_1->INTTYPESET & ch_bit))
+                {
+                    CMSDK_GPIO_1->INTCLEAR = ch_bit;
+                }
+        }
+
+        // Edge trigger
+        if (obj->ch <16)
+        {
+                CMSDK_GPIO_0->INTTYPESET &= ch_bit;
+                if (event == IRQ_RISE) {
+                        CMSDK_GPIO_0->INTPOLSET |= ch_bit;
+                        if (enable) {
+                                CMSDK_GPIO_0->INTENSET |= ch_bit;
+                        } else {
+                                CMSDK_GPIO_0->INTENCLR |= ch_bit;
+                        }
+                } else {
+                        CMSDK_GPIO_0->INTPOLCLR |= ch_bit;
+                        if (enable) {
+                                CMSDK_GPIO_0->INTENSET |= ch_bit;
```

Comment:
fix the formatting in this file, 4 spaces, ``if () {`` etc, one statement per line (not ``case: do-sth(); break;``)